packages/sit-onyx/src/components/OnyxDataGrid/OnyxDataGrid.vue

```diff
@@ -41,10 +41,10 @@ defineSlots<{
 }>();
 
 // Using Ref types to avoid `UnwrapRef` issues
-const renderColumns: Ref<DataGridRendererColumn<TEntry>[]> = ref([]);
-const renderRows: Ref<DataGridRendererRow<TEntry, DataGridMetadata>[]> = ref([]);
-const rendererColumnGroups: Ref<TableColumnGroup[] | undefined> = ref();
-const rendererScrollContainerAttributes: Ref<HTMLAttributes | undefined> = ref();
+const renderColumns: Ref<DataGridRendererColumn<TEntry>[]> = shallowRef([]);
+const renderRows: Ref<DataGridRendererRow<TEntry, DataGridMetadata>[]> = shallowRef([]);
+const rendererColumnGroups: Ref<TableColumnGroup[] | undefined> = shallowRef();
+const rendererScrollContainerAttributes: Ref<HTMLAttributes | undefined> = shallowRef();
```

Comment:
```suggestion
const renderColumns: ShallowRef<DataGridRendererColumn<TEntry>[]> = shallowRef([]);
const renderRows: ShallowRef<DataGridRendererRow<TEntry, DataGridMetadata>[]> = shallowRef([]);
const rendererColumnGroups: ShallowRef<TableColumnGroup[] | undefined> = shallowRef();
const rendererScrollContainerAttributes: ShallowRef<HTMLAttributes | undefined> = shallowRef();
```

Are we even able to get rid of those explicit types?

Reply:
yes :)